spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :feature do
+  describe 'index page' do
+    before(:each) do
+      @user = User.create(
+        name: 'Kidist',
+        photo: 'https://unsplash.com/photos',
+        bio: 'Teacher from Mexico.',
+        post_counter: 0
+      )
+      @post1 = Post.create(
+        id: rand(1000),
+        author_id: @user,
+        title: 'Hello',
+        text: 'This is my first post',
+        comments_counter: 0,
+        likes_counter: 0
+      )
+      Comment.create(author_id: @user, post: @post1, text: 'This is my first comment')
+      @user = User.all
+      visit user_posts_path(@user.ids)
+    end
+
+    it 'should see the username of all other users' do
+      @user.each do |user|
+        expect(user.name).to have_content('Kidist')
+      end
+    end
+
+    it 'should see profile picture for each user' do
+      @user.each do |user|
+        expect(page).to have_css("img[src*='#{user.photo}']")
+      end
+    end
+
+    it 'should see the number of posts each user has written' do
+      expect(page).to have_content(@user.first.post_counter)
+    end
+
+    it 'I can see the first comments on a post' do
+      expect(page).to have_content(@post1.comments.first)
+    end
+
+    it 'I can see how many comments a post has.' do
+      expect(page).to have_content(@post1.comments_counter)
+    end
+
+    it 'I can see how many likes a post has.' do
+      expect(page).to have_content(@post1.likes_counter)
+    end
+  end
+end
```

Comment:
- [ ] [REQUIRED] Hello team, some of the tests are missing for this file, kindly add
* Tests that assert you can see a post's title.
* Tests that assert you can see some of the post's body.
* Tests that assert you can see a section for pagination if there are more posts than fit on the view.
* Tests that redirect to the post's show page when you click on a post.

Reply:
Thank you for your insightful feedback and for reviewing my project